routers/repo/issue.go

```diff
@@ -605,6 +606,15 @@ func commentTag(repo *models.Repository, poster *models.User, issue *models.Issu
 
 // ViewIssue render issue view page
 func ViewIssue(ctx *context.Context) {
+	if extIssueUnit, _ := ctx.Repo.Repository.GetUnit(models.UnitTypeExternalTracker); extIssueUnit != nil {
+		if extIssueUnit.ExternalTrackerConfig().ExternalTrackerStyle == markup.IssueNameStyleNumeric {
```

Comment:
I think that this could probably be extended to alphanumeric IDs. Any difficulties about it?

Reply:
It can, but is it wanted? There is no way to refer to alphanumeric issues on other repository, so is there any point in redirecting? 
Also that would mean you can't use `#` to refer to pull requests when you use alphanumeric tracker. That is maybe fine but you will not gain much, unless we solve the referencing....

Reply:
Do you want to look at supporting references of type `UserZ/RepoZ#AAA-1234`?

Reply:
Fair enough 😄

Reply:
I am not against it, just trying to point out the consequences vs benefits.

Reply:
It only seemed artificial to explicitly avoid supporting it in this function. You have a point, though.